demo/src/main/java/com/liulishuo/filedownloader/demo/NotificationSampleActivity.java

```diff
@@ -136,127 +182,96 @@ protected boolean disableNotification(BaseDownloadTask task) {
+                builder = new NotificationCompat.Builder(
+                        FileDownloadHelper.getAppContext(),
+                        channelId);
+            } else {
+                //noinspection deprecation
+                builder = new NotificationCompat.Builder(FileDownloadHelper.getAppContext())
+                        .setDefaults(Notification.DEFAULT_LIGHTS)
+                        .setPriority(NotificationCompat.PRIORITY_MIN);
+            }
 
-            builder.setDefaults(Notification.DEFAULT_LIGHTS)
-                    .setOngoing(true)
-                    .setPriority(NotificationCompat.PRIORITY_MIN)
-                    .setContentTitle(getTitle())
+            builder.setContentTitle(getTitle())
                     .setContentText(desc)
                     .setContentIntent(pendingIntent)
                     .setSmallIcon(R.mipmap.ic_launcher);
-
         }
 
         @Override
         public void show(boolean statusChanged, int status, boolean isShowProgress) {
-
-            String desc = getDesc();
+            String desc = "";
             switch (status) {
                 case FileDownloadStatus.pending:
                     desc += " pending";
+                    builder.setProgress(getTotal(), getSofar(), true);
                     break;
                 case FileDownloadStatus.started:
                     desc += " started";
+                    builder.setProgress(getTotal(), getSofar(), true);
                     break;
                 case FileDownloadStatus.progress:
                     desc += " progress";
+                    builder.setProgress(getTotal(), getSofar(), getTotal() <= 0);
                     break;
                 case FileDownloadStatus.retry:
                     desc += " retry";
+                    builder.setProgress(getTotal(), getSofar(), true);
                     break;
                 case FileDownloadStatus.error:
                     desc += " error";
+                    builder.setProgress(getTotal(), getSofar(), false);
                     break;
                 case FileDownloadStatus.paused:
                     desc += " paused";
+                    builder.setProgress(getTotal(), getSofar(), false);
                     break;
                 case FileDownloadStatus.completed:
                     desc += " completed";
+                    builder.setProgress(getTotal(), getSofar(), false);
                     break;
                 case FileDownloadStatus.warn:
                     desc += " warn";
+                    builder.setProgress(0, 0, true);
                     break;
             }
 
-            builder.setContentTitle(getTitle())
-                    .setContentText(desc);
-
-
-            if (statusChanged) {
-                builder.setTicker(desc);
-            }
-
-            builder.setProgress(getTotal(), getSofar(), !isShowProgress);
+            builder.setContentTitle(getTitle()).setContentText(desc);
```

Comment:
why remove `setTicker` operation when the status changed?

Reply:
I think the "ticker" text which is sent to accessibility services is unnecessary for our sample.